tests/fe/fe_enriched.cc

```diff
@@ -0,0 +1,794 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// basic checks of FE_Enriched class (see description for each small subtest)
```

Comment:
This is a *really* long test. If something goes wrong, it will be quite difficult to see right away which part of the test it is that broke. Consequently, we have gradually moved away from such large tests and just split them into many smaller ones that all only test one individual aspect of a class. If they share significant code, you can always put the shared code (and/or `main` function) into a `.h` file that you `#include` from all of the tests. There are numerous examples of this style in the testsuite.